lib/rubycritic/analysers/smells/reek.rb

```diff
@@ -1,22 +1,21 @@
 # frozen_string_literal: true
 require 'rubycritic/analysers/helpers/reek'
 require 'rubycritic/core/smell'
-require 'rubycritic/colorize'
 
 module RubyCritic
   module Analyser
     class ReekSmells
-      include Colorize
-      def initialize(analysed_modules)
+      def initialize(analysed_modules, logger=nil)
         @analysed_modules = analysed_modules
+        @logger = logger
```

Comment:
Now that is my biggest problem with this PR. This line and the constructor line and the line below are repeated in every analyzer. So like 5 or 6 times which is a pretty big smell in my mind. 

We either should use a new module named "Loggable" here or - and that would be my preferred choice - create a "BaseAnalyzer" from which all analyzers inherit. This would allow us to remove most, if not all the duplication.